File: frontend/src/types/career.ts
export interface CareerPathSummary {
  id: string
  name: string
  slug: string
  description: string | null
}

export interface CareerPathSkill {
  skill_id: string
  skill_name: string
  skill_slug: string
  weight: number
}

export interface CareerPathDetail extends CareerPathSummary {
  skills: CareerPathSkill[]
}

export interface SkillProgress {
  skill_id: string
  skill_name: string
  skill_slug: string
  weight: number
  lessons_completed: number
  lessons_total: number
  completion: number
}

export interface CareerPathProgress {
  career_path_id: string
  career_path_name: string
  overall_completion: number
  skills: SkillProgress[]
}

export type LearningLevel = 'beginner' | 'practical' | 'technical' | 'advanced' | 'professional'

export interface InterviewQuestion {
  id: string
  question: string
  category: string
  difficulty: LearningLevel
  sample_answer: string
  career_path_id: string | null
}

File: backend/app/schemas/career.py
import uuid

from pydantic import BaseModel, ConfigDict

from app.models.base import LearningLevel


class CareerPathSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    name: str
    slug: str
    description: str | None


class CareerPathSkillRead(BaseModel):
    skill_id: uuid.UUID
    skill_name: str
    skill_slug: str
    weight: float


class CareerPathDetail(CareerPathSummary):
    skills: list[CareerPathSkillRead]


class SkillProgress(BaseModel):
    skill_id: uuid.UUID
    skill_name: str
    skill_slug: str
    weight: float
    lessons_completed: int
    lessons_total: int
    completion: float


class CareerPathProgress(BaseModel):
    career_path_id: uuid.UUID
    career_path_name: str
    overall_completion: float
    skills: list[SkillProgress]


class InterviewQuestionRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    question: str
    category: str
    difficulty: LearningLevel
    sample_answer: str
    career_path_id: uuid.UUID | None
